### DISCUSSION TITLE
Collapse consecutive keypress events in the events sidepanel

### DISCUSSION THREAD

**Discussion by jaril** [RecordReplay/devtools#4622] [Ideas]
Follow up from #3307.

Add heuristics for collapsing keyboard events that should be together. Instead of having `Key Press H` and `Key Press I`, there should be some logic that realizes that perhaps the user was typing something and it should be presented as `Typed HI`.

**Comment by jcmorrow** [comment]
I had originally picked this up for this week, starting with #3994  as a precursor. After talking with @jonbell-lot23 I'm going to punt on implementing this for right now. Jon's concerns are centered around losing the mapping of 1:1 between user events and elements in the Events Timeline. It's a trade-off between verbosity and clarity. Our current view is extremely clear, at the expense of being *very* verbose.

I suspect that we might be able to take some inspiration from a directory structure to make this more manageable, so for a user who typed "Hello" they might see:

```
Typed "Hello World"
    ├── Pressed Key "H"
    │   ├── Key Down "H"
    │   ├── Key Press "H"
    │   ├── Key Up "H"
    ├── Pressed Key "e"
    ├── Pressed Key "l"
    ├── Pressed Key "l"
    ├── Pressed Key "o"
```

This might allow the Replay user to drill down into the specific "Key Up on `H`" that caused their issue, without having to sort through a million events to find it. There's definitely some potential here, but we agreed it would be best implemented when we had a better grasp on how this tool is most useful. Also, as a temporary solution (or at least partial solution) to the verbosity problem, we'll be reducing the padding on events soon, so at least it will be easier to see the whole list at once.

**Comment by jasonLaster** [comment]
Nice I think some form of tree view makes sense in the future. Today, it's probably fine to just simplify what we're showing the user while keeping it literal.

I think we just need a subset of the events we show now. For instance, I think it's fine to pluck the first of n events so the user seeks `Key Down "H"` and not Down, Press, Up. This should both be simpler and still literal. 

Down the line we can abstract it more and more so it's easier to follow

**Comment by jonbell-lot23** [comment]
I agree with everything Jason just said

**Comment by jasonLaster** [comment]
Moving to a discussion since this is not something we will implement right now and it will be nice to get more thoughts from the community
